src/set_theory/game/pgame.lean

```diff
@@ -1231,6 +1240,29 @@ instance covariant_class_add_lt : covariant_class pgame pgame (+) (<) :=
   exact ⟨add_le_add_left h.1 x, add_lf_add_left h.2 x⟩
 end⟩
 
+theorem add_lf_add_of_lf_of_lt {w x y z : pgame.{u}} (hwx : w ⧏ x) (hyz : y < z) : w + y ⧏ x + z :=
```

Comment:
Here's a stronger result and shorter proof for you:
```
theorem add_lf_add_of_lf_of_le {w x y z : pgame.{u}} (hwx : w ⧏ x) (hyz : y ≤ z) : w + y ⧏ x + z :=
lf_of_lf_of_le (add_lf_add_right hwx y) (add_le_add_left hyz x)
```
I wonder whether we should add all the covariant and contravariant classes for all three relations (<), (⧏) and (≤) and all three operations (+), `swap (+)`, and `swap (-)`. If casting to `game` works well we may not need those.

Reply:
I've been told that there's no point in the contravariant class for `⧏`, since there's only really support for the usual order relations. The other ones are definitely worth adding.

Also, thanks for the golfed proof! No idea how I missed it.

Reply:
I do want to make the refactor where I move `game ` earlier down the file first, since it would make the proofs of these theorems much easier.

Reply:
In fact contravariance of (⧏) is equivalent to covariance of (≤), because of [contraposition](https://leanprover-community.github.io/mathlib_docs/logic/basic.html#not_imp_not) ...